File: Frontend/src/events/converters/auctions.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { AuctionsDTO } from "@/events/dto/auctions.dto";
import type { Auctions } from "@/models";

export class AuctionsConverter {
  public static convert(auctionsDto: DeepRequired<AuctionsDTO>): Auctions {
    return {
      divineBarrier: auctionsDto.divineBarrier,
      hazardShield: auctionsDto.hazardShield,
      sniperShield: auctionsDto.sniperShield,
      dramonShield: auctionsDto.dramonShield,
      yinYangWand: auctionsDto.yinYangWand,
    };
  }
}

File: Backend/Events/Converters/AuctionsConverter.cs
using Backend.Domain.Models;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class AuctionsConverter
{
    public static AuctionsDTO ToDTO(Auctions auctions)
    {
        return new AuctionsDTO
        {
            DivineBarrier = auctions.DivineBarrier,
            HazardShield = auctions.HazardShield,
            SniperShield = auctions.SniperShield,
            DramonShield = auctions.DramonShield,
            YinYangWand = auctions.YinYangWand
        };
    }
}
